server/src/main/java/org/opensearch/node/remotestore/RemoteStorePinnedTimestampService.java

```diff
@@ -0,0 +1,272 @@
+
+    private final Supplier<RepositoriesService> repositoriesService;
+    private final Settings settings;
+    private final ThreadPool threadPool;
+    private final ClusterService clusterService;
+    private BlobStoreRepository blobStoreRepository;
+    private BlobStoreTransferService blobStoreTransferService;
+    private RemoteStorePinnedTimestampsBlobStore pinnedTimestampsBlobStore;
+    private AsyncUpdatePinnedTimestampTask asyncUpdatePinnedTimestampTask;
+    private volatile TimeValue pinnedTimestampsSchedulerInterval;
+
+    /**
+     * Controls pinned timestamp scheduler interval
+     */
+    public static final Setting<TimeValue> CLUSTER_REMOTE_STORE_PINNED_TIMESTAMP_SCHEDULER_INTERVAL = Setting.timeSetting(
+        "cluster.remote_store.pinned_timestamps.scheduler_interval",
+        TimeValue.timeValueMinutes(3),
+        TimeValue.timeValueMinutes(1),
+        Setting.Property.NodeScope
+    );
+
+    public RemoteStorePinnedTimestampService(
+        Supplier<RepositoriesService> repositoriesService,
+        Settings settings,
+        ThreadPool threadPool,
+        ClusterService clusterService
+    ) {
+        this.repositoriesService = repositoriesService;
+        this.settings = settings;
+        this.threadPool = threadPool;
+        this.clusterService = clusterService;
+
+        pinnedTimestampsSchedulerInterval = CLUSTER_REMOTE_STORE_PINNED_TIMESTAMP_SCHEDULER_INTERVAL.get(settings);
+    }
+
+    /**
+     * Starts the RemoteStorePinnedTimestampService.
+     * This method validates the remote store configuration, initializes components,
+     * and starts the asynchronous update task.
+     */
+    public void start() {
+        validateRemoteStoreConfiguration();
+        initializeComponents();
+        startAsyncUpdateTask();
+    }
+
+    private void validateRemoteStoreConfiguration() {
+        final String remoteStoreRepo = settings.get(
+            Node.NODE_ATTRIBUTES.getKey() + RemoteStoreNodeAttribute.REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY
+        );
+        assert remoteStoreRepo != null : "Remote Cluster State repository is not configured";
+        final Repository repository = repositoriesService.get().repository(remoteStoreRepo);
+        assert repository instanceof BlobStoreRepository : "Repository should be instance of BlobStoreRepository";
+        blobStoreRepository = (BlobStoreRepository) repository;
+    }
+
+    private void initializeComponents() {
+        String clusterName = ClusterName.CLUSTER_NAME_SETTING.get(settings).value();
+        blobStoreTransferService = new BlobStoreTransferService(blobStoreRepository.blobStore(), this.threadPool);
+        pinnedTimestampsBlobStore = new RemoteStorePinnedTimestampsBlobStore(
+            blobStoreTransferService,
+            blobStoreRepository,
+            clusterName,
+            this.threadPool,
+            ThreadPool.Names.REMOTE_STATE_READ
```

Comment:
will we be using the same threadpool or plans to add a new one later?

Reply:
Ye, for now, same threadpool.